### PR TITLE
chore: DIP4 Adjusted Coinbase description to v20

### PR DESCRIPTION
**Pull request by ogabrielides**
Added changes brought by DIP-29 and DIP-27 to Coinbase.

### CODE REVIEW

File: dip-0004.md [hunk 1]
```diff
@@ -62,6 +62,17 @@ Starting with version >= 2, the following fields are added:
...
+Starting with version >= 3, the following fields are added:
+
+| Field | Type | Size | Description |
+| ----- | ---- | ---- | ----------- |
+| bestCLHeightDiff | varint | 4 | Number of blocks between the current block and the block locked with `bestCLSignature`. |
+| bestCLSignature | BLSSig | 96 | Newest ChainLock signature known by the miner. |
+| creditPoolBalance | int64_t | 8 | Total amount of locked credits at the block height. |
+
+More information on the fields `bestCLHeightDiff` and `bestCLSignature` can be found in [DIP0029 - Randomness Beacon For LLMQ Selection](https://github.com/dashpay/dips/blob/master/dip-0029.md#change-to-the-coinbase-transaction).
+More information on the field `creditPoolBalance` can be found in [DIP0027 - Platform Credit Funding and Withdrawals](https://github.com/dashpay/dips/blob/master/dip-0027.md).
```
Comment (thephez) [inline dip-0004.md:74]: Let's leave out a link for now and add it once it's available.
```suggestion
More information on the `creditPoolBalance` field will be described in a future document.
```

**Review by thephez — CHANGES_REQUESTED** [review]

File: dip-0004.md [hunk 2]
```diff
@@ -62,6 +62,17 @@ Starting with version >= 2, the following fields are added:
 | ----- | ---- | ---- | ----------- |
 | merkleRootQuorums | uint256 | 32 | Merkle root of currently active LLMQs |
 
+Starting with version >= 3, the following fields are added:
+
+| Field | Type | Size | Description |
+| ----- | ---- | ---- | ----------- |
+| bestCLHeightDiff | compactSize uint | 1-9 | Number of blocks between the current block and the block locked with `bestCLSignature`. |
+| bestCLSignature | BLSSig | 96 | Newest ChainLock signature known by the miner. |
+| creditPoolBalance | int64_t | 8 | Total amount Dash Platform credits locked at the block height. |
```
Comment (thephez) [inline dip-0004.md:71]: This balance is denominated in DASH so maybe smth like this?
```suggestion
| creditPoolBalance | int64_t | 8 | Total amount DASH currently locked as of this block height. |
```

File: dip-0004.md [hunk 3]
```diff
@@ -62,6 +62,17 @@ Starting with version >= 2, the following fields are added:
 | ----- | ---- | ---- | ----------- |
 | merkleRootQuorums | uint256 | 32 | Merkle root of currently active LLMQs |
 
+Starting with version >= 3, the following fields are added:
+
+| Field | Type | Size | Description |
+| ----- | ---- | ---- | ----------- |
+| bestCLHeightDiff | varint | 4 | Number of blocks between the current block and the block locked with `bestCLSignature`. |
+| bestCLSignature | BLSSig | 96 | Newest ChainLock signature known by the miner. |
+| creditPoolBalance | int64_t | 8 | Total amount of locked credits at the block height. |
```
[the remark this answers is not in the document]
Comment (thephez) [reply dip-0004.md]: Confirmed on slack that this is DASH denominated

**Review by knst — APPROVED** [review]
LGTM

**Review by UdjinM6 — APPROVED** [review]

**Review by thephez — APPROVED** [review]

**Pull request merged**